src/Infrastructure/IO/src/ESMCI_PIO_Handler.C

```diff
@@ -2021,28 +2050,46 @@ int PIO_IODescHandler::constructPioDecomp(
     handle->arrayShape = (int *)NULL;
   }
   handle->arrayShape = new int[handle->arrayRank];
-  for (int i = 0; i < handle->arrayRank; ++i) {
-    handle->arrayShape[i] = (totalUBound[(tile * handle->arrayRank) + i] -
-                             totalLBound[(tile * handle->arrayRank) + i] +
-                             1);
```

Comment:
Note that I changed the calculation of arrayShape here: I'm pretty sure it is incorrect to have tile in here: this is documented as having dimension [redDimCount*ssiLocalDeCount]. So in principle we should have the DE number in here, but we can leave that out as long as we assume no more than one DE per PE.